userd/settings.go

```diff
@@ -0,0 +1,222 @@
+}
+
+// Settings implements the 'io.snapcraft.Settings' DBus interface.
+type Settings struct {
+	conn *dbus.Conn
+}
+
+// Name returns the name of the interface this object implements
+func (s *Settings) Name() string {
+	return "io.snapcraft.Settings"
+}
+
+// BasePath returns the base path of the object
+func (s *Settings) BasePath() dbus.ObjectPath {
+	return "/io/snapcraft/Settings"
+}
+
+// IntrospectionData gives the XML formatted introspection description
+// of the DBus service.
+func (s *Settings) IntrospectionData() string {
+	return settingsIntrospectionXML
+}
+
+// some notes:
+// - we only set/get desktop files
+// - all desktop files of snaps are prefixed with: ${snap}_
+// - on get/check/set we need to add/strip this prefix
+
+// Check implements the 'Check' method of the 'io.snapcraft.Settings'
+// DBus interface.
+//
+// Example usage: dbus-send --session --dest=io.snapcraft.Settings --type=method_call --print-reply /io/snapcraft/Settings io.snapcraft.Settings.Check string:'default-web-browser' string:'firefox.desktop'
+func (s *Settings) Check(setting, check string, sender dbus.Sender) (string, *dbus.Error) {
+	// avoid information leak: see https://github.com/snapcore/snapd/pull/4073#discussion_r146682758
+	snap, err := snapFromSender(s.conn, sender)
+	if err != nil {
+		return "", dbus.MakeFailedError(err)
+	}
+	if err := settingWhitelisted(setting); err != nil {
+		return "", err
+	}
+	if !allowedSettingsValues.MatchString(check) {
+		return "", dbus.MakeFailedError(fmt.Errorf("cannot check setting %q to value %q: value not allowed", setting, check))
+	}
+
+	// FIXME: this works only for desktop files
+	desktopFile := fmt.Sprintf("%s_%s", snap, check)
+
+	cmd := exec.Command("xdg-settings", "check", setting, desktopFile)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", dbus.MakeFailedError(fmt.Errorf("cannot check setting %s: %s", setting, osutil.OutputErr(output, err)))
+	}
+
+	return strings.TrimSpace(string(output)), nil
+}
+
+// Get implements the 'Get' method of the 'io.snapcraft.Settings'
+// DBus interface.
+//
+// Example usage: dbus-send --session --dest=io.snapcraft.Settings --type=method_call --print-reply /io/snapcraft/Settings io.snapcraft.Settings.Get string:'default-web-browser'
+func (s *Settings) Get(setting string, sender dbus.Sender) (string, *dbus.Error) {
+	if err := settingWhitelisted(setting); err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("xdg-settings", "get", setting)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", dbus.MakeFailedError(fmt.Errorf("cannot get setting %s: %s", setting, osutil.OutputErr(output, err)))
+	}
+
+	// avoid information leak: see https://github.com/snapcore/snapd/pull/4073#discussion_r146682758
+	snap, err := snapFromSender(s.conn, sender)
+	if err != nil {
+		return "", dbus.MakeFailedError(err)
+	}
+	if !strings.HasPrefix(string(output), snap+"_") {
+		return "NOT_THIS_SNAP.snap.desktop", nil
```

Comment:
I wonder if this string would be better: `NOT-THIS-SNAP.desktop`. In this manner, it follows the same pattern as an allowed result (ie `<command>.desktop`, see below) which might help consumers with parsing the result.

Thank you for implementing this check! :)